inframanager/api_handler/api_handler.go

```diff
@@ -1382,32 +1407,97 @@ func (s *ApiServer) SetupHostInterface(ctx context.Context, in *proto.SetupHostI
 
 	logger.Debugf("Interface: %s, port id: %d", in.IfName, portID)
 
-	if status, err := insertRule(s.log, ctx, server.p4RtC, macAddr,
-		ipAddr, portID, p4.HOST); err != nil {
-		logger.Errorf("Failed to insert rule to the pipeline ip: %s mac: %s port id: %d err: %v",
-			ipAddr, macAddr, portID, err)
-		out.Successful = status
-		return out, err
+	/*
+		The inframanger has received the host interface information
+		for the first time. Update the pipeline with the entries
+	*/
+
+	if len(hostInterface.Mac) == 0 {
+		updateHostIP = true
+		updateNodeIP = true
 	}
-	hostInterfaceMac = macAddr
-	store.SetHostInterfaceMac(macAddr)
 
-	if len(config.NodeIP) == 0 {
-		logger.Errorf("No node ip address configured")
-		err = fmt.Errorf("No node ip address configured")
-		out.Successful = false
-		return out, err
+	/*
+		If the infraagent has been restarted for some reason,
+		the inframanager receives the request again. Check and match with
+		the new entries, delete and update the new ip and mac.
+	*/
+	if len(hostInterface.Ip) != 0 && hostInterface.Ip != ipAddr {
+		logger.Infof("Host Interface ip has changed")
+
+		ep := store.EndPoint{
```

Comment:
What are we doing here? If hostInterface.IP is not same as new value received, we are getting an ep from the store with new ipAddress and then getting an entry with new IP address and then deleting it?

Reply:
You are right. Fixed in the next commit.